lamindb/_record.py

```diff
@@ -737,13 +737,19 @@ def save(self, *args, **kwargs) -> Record:
                 revises._revises = None  # ensure we don't start a recursion
                 revises.save()
                 check_name_change(self)
-                super(Record, self).save(*args, **kwargs)
+                if isinstance(self, BasicRecord):  # TODO refactor
+                    super(BasicRecord, self).save(*args, **kwargs)
+                else:
+                    super(Record, self).save(*args, **kwargs)
                 _store_record_old_name(self)
             self._revises = None
         # save unversioned record
         else:
             check_name_change(self)
-            super(Record, self).save(*args, **kwargs)
+            if isinstance(self, BasicRecord):
+                super(BasicRecord, self).save(*args, **kwargs)
```

Comment:
I see that `Record` is a subclass of `BasicRecord`, so the first condition will be always triggered. Is it fine? Anyways the else looks redundant here.

Reply:
This is the weirdest behavior here; it has to be eliminated in the ditch-monkey-patching refactor. We'll then use `super()` instead.

For whatever reason this is the only way I could get everything to work.

Reply:
I think you had this problem before you made `Record` a subclass of `BasicRecord`. I think it should work now just fine with `super(BasicRecord, self).save(*args, **kwargs)` only.

Reply:
Because i believe here `isinstance(self, BasicRecord)` always evaluates to `True` for `self` of both `Record` and `BasicRecord`.

Reply:
You're right I can very like switch this out.